frontend-project/src/models/tags.ts

```diff
@@ -23,6 +27,30 @@ const TagsModel: TagModelType = {
   namespace: 'tags',
   state: defaultTagsState,
   effects: {
+    *create({ payload }, { call }) {
+      try {
+        const response = yield call(create, payload);
+        openNotificationWithIcon(
+          'success',
+          formatMessage({ id: 'tags-new.page-create-notiffy-success' }) + response.data.id,
+        );
+        router.replace(`/tags/`);
+      } catch (err) {
+        if (err.response.status === 400 && err.response.body.name) {
+          err.response.body.name.forEach(message =>
```

Comment:
Czy możemy to zapisać generycznie, aby błędy w polach wyświetlały się przy polach formularzach, zamiast w powiadomieniu bez informacji jakiego pola dotyczy błąd?

Reply:
Ja nie widzę problemu, aby dopracować strukturę odpowiedzi błędu API, ale potrzebne wytyczne co do potrzeb.

Reply:
@ad-m  W oknie logowania, mamy wyświetlanie problemu w formularzu[ (pomiedzy "Hasło logowania do konta" a polem z loginem )](https://github.com/watchdogpolska/small_eod/blob/7d172428f3aee4c9003ce995956b7391716bb98d/frontend-project/src/pages/user/login/index.tsx#L52) I gdybym posiedział to jestem w stanie jeszcze to odwzorować. 
Niestety nie wiem jak mogła by wyglądać struktura odpowiedzi z api aby przypisywać błąd do konkretnego pola w formularzu. Pozwolę sobie zawołać @kuskoman, czy jesteś w stanie coś tutaj zasugerować ?

Reply:
Obecna odpowiedź API wskazuje konkretne pole, bo masz `name`. Ja proponuje starać się użyć obecnej odpowiedzi, a jak problem zostanie rozpracowany mocniej (więcej przypadków) to zmienimy strukturę API, aby uprościć implementacje w front-endzie. Wykonamy to przed wejściem z V2 na produkcje.

Reply:
@ad-m siedziałem trochę godzin nad zagadnieniem, próbowałem szukać pomocy na forach jednak nie udało mi się zrobić odpowiedzi według Twoich założeń. Dokładniej mam problem z przekazaniem odpowiedzi z Modelu do formularza, próbowałem odgapić to z elementu logowania bądź z Cases/New (ponieważ są tam zwracane kolekcje tagów/instytucji itd.) jednak nic z tego. 
Odniosę się tutaj do reszty issue, chętnie się za nie wezmę gdy ktoś opracuje standard edycji/dodawania/usuwania ponieważ tak jak mówiłem na pierwszym call'u to moje początki z reactem i reduxem i dopiero poznaje mechanikę ich działania :)

Reply:
Ja w zakresie Reacta nie mam zbytniego doświadczenia także, ale podglądając zamykając projekt w Vue, z którym mam zaproponowałem:

```javascript
const handleSubmit = (value: Tag) => {
    // dispatch({
    //   type: 'tags/create',
    //   payload: { ...value },
    // });
    const response = {
      name: [`To pole jest wymagane ${Math.random()}.`]
    };
    form.setFields(Object
        .entries(response)
        .map(([name, errors]) => ({ name, errors }))
    );
};
```

Teraz pozostaje tylko rozważenie gdzie ta logika obsługi błędów będzie się znajdować, czy w komponencie strony (jak teraz wkleiłem), czy w modelu (jak teraz jest), czy w serwisie (jak mogłoby teoretycznie być), a następnie odpowiednie przekazanie danych. Proszę o komentarz @fervero i @mateuszmagulski , którzy mają większe doświadczenie w projekcie.
 
Jeżeli będziemy mieli ten element poprawnie rozwiązany to możemy znacznie więcej walidacji pchnąć na back-end (na początku nawet nie musimy tracić czasu na komunikat dla wymaganego pola, bo back-end to sprawdzi). W zespole obecnie nie widzę osoby, która po prostu może wyznaczyć standard, więc raczej nie unikniemy tego.

Reply:
<img width="1224" alt="Zrzut ekranu 2020-10-26 o 19 18 03" src="https://user-images.githubusercontent.com/9075439/97211913-0598ed00-17c0-11eb-83a3-21c1d90d6dad.png">
@ad-m Ale taka walidacja na próbę zapisu pustego pola to była...

Reply:
Ale mój kod nie odnosi się wyłącznie do wymaganego pola :) Kod może przetwarzać także pozostałe błędy z back-endu (nieprawidłowy format, złą długość itd.). Poza tym jest gwarantowane, że będzie to aktualne. Akurat tym razem przypadek nam się powtórzył, ale patrz szerzej jako na standard.

Reply:
Tylko wąski zakres błędów jesteś w stanie sprawdzić po stronie front-endu.

Reply:
<img width="1434" alt="Zrzut ekranu 2020-10-26 o 22 01 05" src="https://user-images.githubusercontent.com/9075439/97228189-d5107d80-17d6-11eb-9616-3e220cbde46e.png">
Dowiozłem to !